scripts/test-upgrade.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# TAG=<OPTIONAL TAG> ./test-upgrade.sh
+#
+# This script (by default) tests against the `next` branch of the KIC to ensure
+# that chart upgrades work against the latest work being done there.
+#
+# Optionally, the script can be flagged to test upgrades from the current default
+# tag (provided by the chart and values.yaml) to any tag specified.
+# ------------------------------------------------------------------------------
+
+# ------------------------------------------------------------------------------
+# Environment Variables
+# ------------------------------------------------------------------------------
+
+TEST_ENV_NAME="${TEST_ENV_NAME:-kong-charts-tests}"
+TAG="${TAG:-next-railgun}"
+RELEASE_NAME="${RELEASE_NAME:-chart-tests-upgrade-compat}"
+RELEASE_NAMESPACE="${RELEASE_NAMESPACE:-$(uuidgen)}"
+
+# ------------------------------------------------------------------------------
+# Shell Configuration
+# ------------------------------------------------------------------------------
+
+set -euo pipefail
+
+# ------------------------------------------------------------------------------
+# Kubernetes Cluster Setup
+# ------------------------------------------------------------------------------
+
+kubectl cluster-info --context kind-${TEST_ENV_NAME} 1>/dev/null
+KUBERNETES_VERSION="$(kubectl version -o json | jq -r '.serverVersion.gitVersion')"
+
+# ------------------------------------------------------------------------------
+# Setup Chart Cleanup - Kubernetes Ingress Controller
+# ------------------------------------------------------------------------------
+
+function cleanup() {
+    echo "INFO: cleaning up helm release ${RELEASE_NAME}"
+    helm delete --namespace ${RELEASE_NAMESPACE} ${RELEASE_NAME}
+    kubectl delete namespace ${RELEASE_NAMESPACE}
```

Comment:
Long-running cluster takes forever removing namespace, you might want to ensure namespace is cleaned before next step.

Reply:
I do ensure namespace is cleaned up by running the `helm delete` above.

Reply:
? if so, why use duplicate command.